From c93d4273a384373220cb3ab7ac52a138cfb19db7 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:29:10 -0400
Subject: [alsa] hda/realtek: Fixed single output machine get empty hp sense

Message-id: <140972823573-250-patchset-jkysela@redhat.com>
Patchwork-id: 90950
O-Subject: [RHEL 7.1 PATCH 249/442] ALSA: hda/realtek - Fixed single output machine get empty hp sense
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

If it only has single output of HP out of machine.
The driver parser will copy hp_pins to line_out_pins.
hp_pins will empty for alc283_init and alc283_shutup functions.
This will cause not have value for hp_pin_sense.
Add check line_out_type code will solve it .

Signed-off-by: Kailang Yang <kailang@realtek.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 8314f22589297bdb100788112f901fe885edacbc)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index ce564e1..6036529 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -3026,6 +3026,11 @@ static void alc283_init(struct hda_codec *codec)
  bool hp_pin_sense;
  int val;
 
+ if (!spec->gen.autocfg.hp_outs) {
+  if (spec->gen.autocfg.line_out_type == AC_JACK_HP_OUT)
+   hp_pin = spec->gen.autocfg.line_out_pins[0];
+ }
+
  alc283_restore_default_value(codec);
 
  if (!hp_pin)
@@ -3062,6 +3067,11 @@ static void alc283_shutup(struct hda_codec *codec)
  bool hp_pin_sense;
  int val;
 
+ if (!spec->gen.autocfg.hp_outs) {
+  if (spec->gen.autocfg.line_out_type == AC_JACK_HP_OUT)
+   hp_pin = spec->gen.autocfg.line_out_pins[0];
+ }
+
  if (!hp_pin) {
   alc269_shutup(codec);
   return;
-- 
1.7.1